File: backend/Model/Company.cs
namespace YLWorks.Model
{
    public enum CompanyType
    {
        Own = 1,
        Client = 2,
        Supplier = 3
    }

    public class Company : BaseEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;

        public Guid? BillingAddressId { get; set; }
        public Address? BillingAddress { get; set; }

        public Guid? DeliveryAddressId { get; set; }
        public Address? DeliveryAddress { get; set; }

        public string? ContactNo { get; set; }
        public string? ContactPerson1 { get; set; }
        public string? ContactPerson2 { get; set; }
        public string? Email { get; set; }

        public string? FaxNo { get; set; }
        public string? ACNo { get; set; }
        public string? WebsiteUrl { get; set; }
        public string? TINNo { get; set; }
        public string? SSTRegNo { get; set; }
        public CompanyType Type { get; set; }
        public decimal BalancePayment { get; set; } = 0;
        public bool SameAsBillingAddress { get; set; }

        public bool IsActive { get; set; } = true;

        public string? LogoImage { get; set; }
    }

    public class CreateCompanyRequest
    {
        public string Name { get; set; } = string.Empty;

        public AddressRequest? BillingAddress { get; set; }
        public AddressRequest? DeliveryAddress { get; set; }

        public string? ContactNo { get; set; }
        public string? ContactPerson1 { get; set; }
        public string? ContactPerson2 { get; set; }
        public string? FaxNo { get; set; }
        public string? ACNo { get; set; }
        public string? Email { get; set; }
        public string? WebsiteUrl { get; set; }
        public string? TINNo { get; set; }
        public string? SSTRegNo { get; set; }
        public CompanyType Type { get; set; }
        public bool SameAsBillingAddress { get; set; }

        public string? LogoImage { get; set; }

    }

    public class UpdateCompanyRequest : CreateCompanyRequest
    {
        public Guid Id { get; set; }
      
    }
    public class CompanyDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string ContactNo { get; set; }
        public string ContactPerson1 { get; set; }
        public string ContactPerson2 { get; set; }
        public string FaxNo { get; set; }
        public string ACNo { get; set; }
        public string Email { get; set; }
        public string WebsiteUrl { get; set; }
        public CompanyType Type { get; set; }
        public bool IsActive { get; set; }
        public string LogoImage { get; set; }
        public string TINNo { get; set; }
        public string SSTRegNo { get; set; }
        public bool SameAsBillingAddress { get; set; }

        public AddressDto? BillingAddress { get; set; }
        public AddressDto? DeliveryAddress { get; set; }
    }

}

File: frontend/src/app/models/Company.ts
import { CompanyType } from '../shared/enum/enum';
import { AddressDto, AddressRequest } from './Address';
import { BaseModel } from './BaseModel';

export interface CompanyDto extends BaseModel {
  name: string;
  billingAddressId?: string;
  billingAddress?: AddressDto;
  deliveryAddressId?: string;
  deliveryAddress?: AddressDto;
  contactNo: string;
  primaryContactPerson: string;
  primaryContactNo: string;
  primaryEmail: string;
  secondaryContactPerson: string;
  secondaryContactNo: string;
  secondaryEmail: string;
  registrationNo: string;
  email: string;
  faxNo: string;
  acNo: string;
  websiteUrl: string;
  type: CompanyType;
  balancePayment: number;
  isActive: boolean;
  logoImage: string;
  tinNo: string;
  sstRegNo: string;
  ssmRegNo: string;
  sameAsBillingAddress: boolean;
}

export interface CreateCompanyRequest {
  name: string;
  billingAddress?: AddressRequest;
  deliveryAddress?: AddressRequest;
  contactNo?: string;
  primaryContactPerson: string;
  primaryContactNo: string;
  primaryEmail: string;
  secondaryContactPerson: string;
  secondaryContactNo: string;
  secondaryEmail: string;
  faxNo?: string;
  acNo?: string;
  email?: string;
  websiteUrl?: string;
  type: CompanyType;
  registrationNo: string;
  logoImage?: string;
  tinNo: string;
  sstRegNo: string;
  ssmRegNo: string;
  sameAsBillingAddress: boolean;
}

export interface UpdateCompanyRequest extends CreateCompanyRequest {
  id: string;
}
